src/util/dataTypes.js

```diff
@@ -11,10 +11,35 @@ function isCSSFunction(value) {
   return IS_CSS_FN.test(value)
 }
 
+// These properties accept a `<dashed-ident>` as one of the values. This means that you can use them
+// as: `timeline-scope: --tl;`
+//
+// Without the `var(--tl)`, in these cases we don't want to normalize the value, and you should add
+// the `var()` yourself.
+//
+// More info:
+// - https://drafts.csswg.org/scroll-animations/#propdef-timeline-scope
+// - https://developer.mozilla.org/en-US/docs/Web/CSS/timeline-scope#dashed-ident
+//
+const AUTO_VAR_INJECTION_EXCEPTIONS = new Set([
+  // Concrete properties
+  'scroll-timeline-name',
+  'timeline-scope',
+  'view-timeline-name',
+
+  // Shorthand properties
+  'scroll-timeline',
+  'animation-timeline',
+  'view-timeline',
+])
+
 // This is not a data type, but rather a function that can normalize the
 // correct values.
-export function normalize(value, isRoot = true) {
-  if (value.startsWith('--')) {
+export function normalize(value, context = null, isRoot = true) {
+  if (
+    (context ? !AUTO_VAR_INJECTION_EXCEPTIONS.has(context.property) : true) &&
```

Comment:
can we split this into it's own var or something. This is just kinda hard to scan.

Reply:
Maybe:

```js
let isVarException = context && AUTO_VAR_INJECTION_EXCEPTIONS.has(context.property)

if (value.startsWith('--') && !isVarException) { /* … */ }
```